numba_cuda/numba/cuda/tests/cudapy/test_inspect.py

```diff
@@ -37,28 +47,47 @@ def foo(x, y):
         # Signature in annotation
         self.assertIn("(float32, int32)", typeanno)
         file.close()
-        # Function name in LLVM
-        llvm = foo.inspect_llvm(sig)
-        self.assertIn("foo", llvm)
 
-        # Kernel in LLVM
-        self.assertIn("define void @", llvm)
-
-        asm = foo.inspect_asm(sig)
-
-        # Function name in PTX
-        self.assertIn("foo", asm)
-        # NVVM inserted comments in PTX
-        self.assertIn("Generated by NVIDIA NVVM Compiler", asm)
+        filecheck = FileCheckKernel(foo)
+        self.assertTrue(filecheck.check_llvm(sig))
+        self.assertTrue(filecheck.check_asm(sig))
```

Comment:
With the suggestion above (https://github.com/NVIDIA/numba-cuda/pull/342/files#r2225538599), I'd envisage this looking like:

```suggestion
        self.check_llvm(kernel, sig)
        self.check_asm(kernel, sig)
```

which is a bit closer to the idioms of the rest of the unittest suite.

Reply:
What do you think of the more verbose names I chose?
```python
        self.assert_filecheck_llvm(foo, sig)
        self.assert_filecheck_asm(foo, sig)
```
Note that I stuck to snake_case naming conventions while the unittest framework uses mixedCase. I am unsure about this decision.